### PR TITLE
Add thread local storage support

### PR DESCRIPTION
**Pull request by bamadhu**
Adding a workaround for the non-existing TLS support.

### CODE REVIEW

File: patches/src/tinycthread.c.patch [hunk 1]
```diff
@@ -24,3 +53,30 @@ index b0ec8e95..4f351096 100644
...
++#endif
++
+ #if defined(_TTHREAD_WIN32_)
+   /* https://stackoverflow.com/questions/12744324/how-to-detach-a-thread-on-windows-c#answer-12746081 */
+   return CloseHandle(thr) != 0 ? thrd_success : thrd_error;
+@@ -704,7 +730,11 @@ int thrd_join(thrd_t thr, int *res)
+ int thrd_sleep(const struct timespec *duration, struct timespec *remaining)
+ {
+ #if !defined(_TTHREAD_WIN32_)
++#ifdef __MVS__
```
Comment (IgorTodorovskiIBM) [inline patches/src/tinycthread.c.patch:75]: Can you use the nanosleep from zoslib here?

**Review by IgorTodorovskiIBM — APPROVED** [review]
Looks fine overall, when this is upstreamed, it may be cleaner to support these macros for the other platforms as well (if possible). That way you will only need to have the ifdef guards in the macro definitions.

**Review by HarithaIBM — APPROVED** [review]

File: patches/src/rdkafka.c.patch [hunk 2]
```diff
@@ -1,16 +1,190 @@
...
++#ifdef __MVS__
++	char *ret = NULL;
++	static pthread_key_t ret_key;
++	static int ret_key_s = 0;
++	size_t size_of_ret = 32;
++#else
+         static RD_TLS char ret[32];
++#endif
+         int idx = err - RD_KAFKA_RESP_ERR__BEGIN;
++GET_KEY_VAL(ret_key,ret_key_s,ret,size_of_ret*sizeof(char))
```
[the remark this answers is not in the document]
Comment (bamadhu) [reply patches/src/rdkafka.c.patch]: @HarithaIBM This is an outdated change, This line doesn't exist.

File: patches/src/rdkafka_op.c.patch [hunk 3]
```diff
@@ -0,0 +1,15 @@
+diff --git a/src/rdkafka_op.c b/src/rdkafka_op.c
+index 34e9e3fd..55a15e14 100644
+--- a/src/rdkafka_op.c
++++ b/src/rdkafka_op.c
+@@ -739,6 +739,10 @@ rd_kafka_op_res_t
+ rd_kafka_op_call(rd_kafka_t *rk, rd_kafka_q_t *rkq, rd_kafka_op_t *rko) {
+         rd_kafka_op_res_t res;
+         rd_assert(rko->rko_op_cb);
++/*madhu RD_TLS*/
```
[the remark this answers is not in the document]
Comment (bamadhu) [reply patches/src/rdkafka_op.c.patch]: This is an outdated change, This line doesn't exist.

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM

**Pull request merged**